modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v2_0_6/UpgradeDDMDataProviderInstance.java

```diff
@@ -218,14 +220,17 @@ private boolean _updateDDMDataProviderRules(JSONArray rulesJSONArray) {
 							CharPool.SEMICOLON);
 					}
 
-					newActionOutputs = newActionOutputs.substring(
-						0, newActionOutputs.length() - 1);
+					if (newActionOutputs.length() > 0) {
+						newActionOutputs = newActionOutputs.substring(
+							0, newActionOutputs.length() - 1);
+					}
 
 					action = StringBundler.concat(
-						"call('", arguments[0], "', '", arguments[1], "', '",
+						"call('", uuid, "', '", arguments[1], "', '",
 						newActionOutputs, "')");
 
-					updated = true;
+					updated =
+						updated || !newActionOutputs.equals(actionOutputs);
```

Comment:
@rodrigopaulino, the variable `updated` isn't modified until this line with the proposed fix, therefore what's the purpose of assign it here? Isn't enough keep only `!newActionOutputs.equals(actionOutputs)`?

Reply:
No, because this way a false result could override a true one.